src/libraries/Common/src/Interop/OSX/Swift.Runtime/CryptoKit.cs

```diff
@@ -0,0 +1,274 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Runtime.InteropServices.Swift;
+
+#pragma warning disable CS3016 // Arrays as attribute arguments are not CLS Compliant
+#pragma warning disable SYSLIB1051
+
+namespace Swift.Runtime
+{
+    /// <summary>
+    /// Represents ChaChaPoly in C#.
+    /// </summary>
+    internal unsafe partial struct ChaChaPoly
+    {
+        /// <summary>
+        /// Represents Nonce in C#.
+        /// </summary>
+        internal sealed unsafe partial class Nonce
+        {
+            private const int _payloadSize = 16;
+            internal void* payload;
+
+            internal Nonce()
+            {
+                payload = Marshal.AllocHGlobal(_payloadSize).ToPointer();
+                SwiftIndirectResult swiftIndirectResult = new SwiftIndirectResult(payload);
+                CryptoKit.PInvoke_ChaChaPoly_Nonce_Init(swiftIndirectResult);
+            }
+
+            internal Nonce(Data data)
+            {
+                payload = Marshal.AllocHGlobal(_payloadSize).ToPointer();
+                SwiftIndirectResult swiftIndirectResult = new SwiftIndirectResult(payload);
+                void* dataPtr = &data;
+
+                CryptoKit.PInvoke_ChaChaPoly_Nonce_Init2(swiftIndirectResult, dataPtr, data.Metadata(), default(DataProtocol).WitnessTable(data), out SwiftError error);
+
+                if (error.Value != null) // Should throw when nonce is not 12 bytes
+                {
+                    throw new Exception();
+                }
+            }
+
+            ~Nonce()
+            {
+                Marshal.FreeHGlobal(new IntPtr(payload));
+            }
+        }
+
+        /// <summary>
+        /// Represents SealedBox in C#.
+        /// </summary>
+        [StructLayout(LayoutKind.Sequential, Size = 16)]
+        internal unsafe partial struct SealedBox
+        {
+            internal Data combined;
```

Comment:
What is "combined"? The cipher text with the tag appended?

Reply:
Tag, nonce and ciphertext. As per https://developer.apple.com/documentation/cryptokit/chachapoly/sealedbox